### PR TITLE
Avoid docker mounting to support CLI commands inside of a docker container

### LINKED ISSUE #1325: Using cortex inside a docker
Hi everyone
We have installed cortex into an alpine docker using `RUN bash -c "$(curl -sS https://raw.githubusercontent.com/cortexlabs/cortex/0.18/get-cli.sh)"`and its installed well, but when we try to connect to our pre-existing cluster using `"cortex cluster info --config my_cluster.yaml -env aws"` it gives this error: 

```
foqum-read-cortex-api_1  | fetching cluster status ...
foqum-read-cortex-api_1  | 
foqum-read-cortex-api_1  | panic: runtime error: invalid memory address or nil pointer dereference [recovered]
foqum-read-cortex-api_1  |      panic: runtime error: invalid memory address or nil pointer dereference
foqum-read-cortex-api_1  | [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x13793f2]
foqum-read-cortex-api_1  | 
foqum-read-cortex-api_1  | goroutine 1 [running]:
foqum-read-cortex-api_1  | github.com/cortexlabs/cortex/pkg/lib/exit.Panic(0x299c940, 0xc000845b40, 0x0, 0x0, 0x0)
foqum-read-cortex-api_1  |      /home/circleci/project/pkg/lib/exit/exit.go:60 +0x1c0
foqum-read-cortex-api_1  | github.com/cortexlabs/cortex/pkg/lib/exit.RecoverAndExit(0x0, 0x0, 0x0)
foqum-read-cortex-api_1  |      /home/circleci/project/pkg/lib/exit/exit.go:66 +0x9d
foqum-read-cortex-api_1  | panic(0x21a2080, 0x3d3e440)
foqum-read-cortex-api_1  |      /usr/local/go/src/runtime/panic.go:969 +0x166
foqum-read-cortex-api_1  | github.com/cortexlabs/cortex/pkg/types/clusterconfig.(*Config).UserTable(0xc00036c390, 0x0, 0x0, 0x0)
foqum-read-cortex-api_1  |      /home/circleci/project/pkg/types/clusterconfig/clusterconfig.go:1032 +0xe2
foqum-read-cortex-api_1  | github.com/cortexlabs/cortex/pkg/types/clusterconfig.(*InternalConfig).UserTable(0xc00036c390, 0x0, 0x0, 0x0)
foqum-read-cortex-api_1  |      /home/circleci/project/pkg/types/clusterconfig/clusterconfig.go:1020 +0xc9
foqum-read-cortex-api_1  | github.com/cortexlabs/cortex/cli/cmd.printInfoClusterConfig(0xc00036c380)
foqum-read-cortex-api_1  |      /home/circleci/project/cli/cmd/cluster.go:501 +0xc6
foqum-read-cortex-api_1  | github.com/cortexlabs/cortex/cli/cmd.printInfoOperatorResponse(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
foqum-read-cortex-api_1  |      /home/circleci/project/cli/cmd/cluster.go:491 +0x220
foqum-read-cortex-api_1  | github.com/cortexlabs/cortex/cli/cmd.cmdInfo(0xc0005659a0, 0x14, 0xc00030fe00, 0x28, 0xc00055e040, 0x14, 0xc00030fe30, 0x28, 0xc00009c140, 0xc00055e000)
foqum-read-cortex-api_1  |      /home/circleci/project/cli/cmd/cluster.go:441 +0x415
foqum-read-cortex-api_1  | github.com/cortexlabs/cortex/cli/cmd.glob..func3(0x3d558e0, 0xc00011d100, 0x0, 0x4)
foqum-read-cortex-api_1  |      /home/circleci/project/cli/cmd/cluster.go:272 +0x1df
foqum-read-cortex-api_1  | github.com/spf13/cobra.(*Command).execute(0x3d558e0, 0xc00011d080, 0x4, 0x4, 0x3d558e0, 0xc00011d080)
foqum-read-cortex-api_1  |      /home/circleci/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:846 +0x29d
foqum-read-cortex-api_1  | github.com/spf13/cobra.(*Command).ExecuteC(0x3d56600, 0xc00069ff40, 0x1, 0x1)
foqum-read-cortex-api_1  |      /home/circleci/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x349
foqum-read-cortex-api_1  | github.com/spf13/cobra.(*Command).Execute(...)
foqum-read-cortex-api_1  |      /home/circleci/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
foqum-read-cortex-api_1  | github.com/cortexlabs/cortex/cli/cmd.Execute()
foqum-read-cortex-api_1  |      /home/circleci/project/cli/cmd/root.go:152 +0x300
foqum-read-cortex-api_1  | main.main()
foqum-read-cortex-api_1  |      /home/circleci/project/cli/main.go:24 +0x20 
```

We have solved this error by mapping our host  `/root/.cortex` to the `/root/.cortex `of our docker, inside the `docker-compose` file.
The thing is, we don't want to have to need to do that mapping . We want to know why after installing the `cortex cli` using the command mentioned at the beginning of this text, the `.cortex` folder is not created directly within our docker `root` folder.
To sum up, when we install the cortex client, the `.cortex` folder is not created

Thanks for advance

### LINKED ISSUE #1361: refreshCachedClusterConfig() can return a cluster config with nil region
#### Description

Likely the root cause of https://github.com/cortexlabs/cortex/issues/1325

One theory is that an issue with the volume mount could cause this, i.e. `refresh_cluster_config.py` executes successfully, but since the cluster config file it writes to is not mounted properly, it doesn't propagate back to the CLI.

### PR DESCRIPTION
**Pull request by deliahu**
closes #1361, closes #1325

---

checklist:

- [x] run `make test` and `make lint`
- [x] test manually (i.e. build/push all images, restart operator, and re-deploy APIs)

### CODE REVIEW

File: cli/local/errors.go [hunk 1]
```diff
@@ -113,6 +114,14 @@ func ErrorPortAlreadyInUse(port int) error {
 func ErrorUnableToFindAvailablePorts() error {
 	return errors.WithStack(&errors.Error{
 		Kind:    ErrUnableToFindAvailablePorts,
-		Message: fmt.Sprintf("unable to find available ports"),
+		Message: "unable to find available ports",
+	})
+}
+
+func ErrorBindDockerInDocker(err error) error {
```
Comment (deliahu) [inline cli/local/errors.go:121]: Here's what the error looks like:

Screen Shot 2020-09-21 at 4 38 32 PM

File: cli/cmd/errors.go [hunk 2]
```diff
@@ -130,6 +131,20 @@ func ErrorInvalidOperatorEndpoint(endpoint string) error {
 	})
 }
 
+// err can be passed in as nil
+func ErrorNoOperatorLoadBalancer(envName string, err error) error {
+	errMsg := fmt.Sprintf("unable to locate operator load balancer; you can attempt to resolve this issue and configure your CLI environment by running `cortex cluster info --env %s`", envName)
+
+	if err != nil {
```
Comment (vishalbollu) [inline cli/cmd/errors.go:138]: This doesn't look like its being used anywhere so I think this code can be removed for now. Also I think that if I saw an invocation of ErrorNoOperatorLoadBalancer() I would expect the return error kind to be `ErrNoOperatorLoadBalancer` but in this case, it's possible that the result is a different error kind with the message modified.. We can discuss this error pattern if you think it is important

File: manager/cluster_config_env.py [hunk 3]
```diff
@@ -15,13 +15,16 @@
...
-    if base_key.lower().startswith("cortex_tags"):
-        inlined_tags = ",".join([f"{k}={v}" for k, v in value.items()])
-        print(f"export CORTEX_TAGS={inlined_tags}")
-        print(f"export CORTEX_TAGS_JSON='{json.dumps(value)}'")
+    if base_key.lower() == "cortex_tags":
+        exportTags(value, "CORTEX_TAGS")
+        exportTags(
+            value, "CORTEX_OPERATOR_LOAD_BALANCER_TAGS", {"cortex.dev/load-balancer": "operator"}
+        )
+        exportTags(value, "CORTEX_APIS_LOAD_BALANCER_TAGS", {"cortex.dev/load-balancer": "apis"})
```
Comment (vishalbollu) [inline manager/cluster_config_env.py:27]: For consistency I think `CORTEX_APIS_LOAD_BALANCER_TAGS` should be renamed to `CORTEX_API_LOAD_BALANCER_TAGS`. Similarly for the value as well.

**Review by vishalbollu — COMMENTED** [review]
Let me know if you want to talk about the comments

File: cli/cmd/errors.go (hunk 2 above)
Comment (deliahu) [reply cli/cmd/errors.go]: Agreed

File: manager/cluster_config_env.py (hunk 3 above)
Comment (deliahu) [reply manager/cluster_config_env.py]: some of the internals use "apis" (e.g. `ingressgateway-apis` and `apis-istio-gateway`), but I agree that "api" makes more sense for the user

**Review by vishalbollu — APPROVED** [review]

**Pull request merged**